test/framework/kubernetes.go

```diff
@@ -139,6 +139,16 @@ func IsPodRunning(pod *corev1.Pod) bool {
 	return pod.Status.Phase == corev1.PodRunning
 }
 
+// IsPodStatusConditionReady returns true if all pod's containers are ready (really running)
+func IsPodStatusConditionReady(pod *corev1.Pod) bool{
```

Comment:
Makes me think - can we replace method `IsPodRunning` with this one? They seem to have same purpose.

Reply:
Can be, this one is more "thorough" I would say as it really returns true only once the pod as a whole (all internal containers) are running. If you check only a Pod Running status, it may be running, but doesn't have to be ready.

Reply:
Switched `CheckPodsAreRunning()` to `CheckPodsAreReady()` and now it calls `IsPodStatusConditionReady()` function. I left `IsPodRunning()` still in the code as it is used for checking if the pod is being monitored via logs. See here: https://github.com/kiegroup/kogito-cloud-operator/blob/2fc81b0c1c1a7888b7642c88f98174c04f781f34/test/framework/logging.go#L142

In this situation, it is enough for a pod to be `Running` as by then all containers should be present I guess and we can get their logs with follow. Wdyt? Waiting for all containers to be `Ready` might not be the best option as it can start producing output even before the start command finished (which only then marks the container as  Ready I guess)?

So let me know what you think about it.

Reply:
Looks good to me, thanks.